include/SH3/arc/vfile.hpp

```diff
@@ -0,0 +1,100 @@
+/** @file
+ *
+ *  Class and data structures describing a Virtual File inside a SILENT HILL 3 arc section.
+ *
+ *  @copyright 2016  Palm Studios
+ *
+ *  @date 5-1-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#ifndef VFILE_HPP_INCLUDED
+#define VFILE_HPP_INCLUDED
+
+#include "SH3/arc/types.hpp"
+
+struct sh3_arc_vfile final
+{
+public:
+    enum class read_result
+    {
+        SUCCESS,
+        END_OF_FILE,
+        PARTIAL_READ,
+    };
+
+    enum origins
+    {
+        ORIGIN_START,
+        ORIGIN_CURRENT,
+    };
+
+    class read_error final
+    {
+    public:
+        operator bool() const { return result != read_result::SUCCESS; }
+        read_result get_read_result() const { return result; }
+
+        void set_error(sh3_arc_vfile::read_result res){result = res;};
+
+        std::string message() const;
+
+    private:
+        read_result result = read_result::SUCCESS;
+    };
+
+
+    sh3_arc_vfile(sh3_arc& mft, std::string fname){Open(mft, fname);fpos = 0;}
+
+
+    /**
+     *  Open a handle to a virtual file.
+     *
+     *  @param mft      The @ref sh3_arc Master File Table, arc.arc.
+     *  @param fname    The name of the file we want to open.
+     *
+     *  @returns @c true if the file was found, @c false if not.
+     */
+    bool Open(sh3_arc& mft, std::string fname);
```

Comment:
`fname` can be `const std::string&`.

Documentation should specify what happens when you call `Open()` on an already opened file.